src/callgraph_mode.ml

```diff
@@ -0,0 +1,24 @@
+open! Core
+
+type t =
+  | Last_branch_record
+  | Frame_pointers
+  | Dwarf
+[@@deriving sexp]
+
+let param =
+  let open Command.Param in
+  flag
+    "-callgraph-mode"
+    (optional (Command.Arg_type.create (fun str -> t_of_sexp (Sexp.of_string str))))
+    ~doc:
+      "When perf is running with sampling collection mode, this sets how perf should \
```

Comment:
In general, try to avoid mentioning perf directly in the help text. Using perf is an "implementation detail", and e.g. the direct backend doesn't shell out to perf, but would have the same constraints (and ideally the help text would continue to apply).

Reply:
Replaced perf with magic-trace.